### ISSUE TITLE
Implement reference secure versions of Node Goad app built using different frameworks

### ISSUE THREAD

**Issue by ckarande** [OWASP/NodeGoat#38]
Build reference secure versions using express, express + passport, MEAN SPA, hapi.js, koa, socket.io, etc

**Comment by binarymist** [comment]
Can you provide some more specifics on this @ckarande ?

**Comment by binarymist** [comment]
bump @ckarande

**Comment by ckarande** [comment]
@binarymist This item on roadmap addresses the question - how the NodeGoat can offer more value as Node.js security learning resource. Here are some thoughts - 

1. I would like to move the security fixes added in the code comments to separate branches. So we would have a branch for the insecure app, and multiple branches (one per OWASP Top 10 vulnerability) with fixes incorporated in it. In the tutorial, we can just embed the diff between branches to show how to fix the vulnerability.

2. Express is obviously the most widely used Node.js framework and current NodeGoat project offers a good amount information on how to secure it. Going forward, we can expand the project to have multiple applications addressing the security aspects of other Node.js server-side frameworks (hapi.js, sails.js, etc depending on their popularity) or architectures (isomorphic apps, or real-time data pushed with web socket, etc).

3. It doesn't need to be just code, we can curate a list of awesome node.js security resources and tools. I can update the issue to reflect the intent.

Does it make sense? Please let me know if you have any suggestions or thoughts.

**Comment by binarymist** [comment]
Yip, I think this is what we discussed in NY yes?

**Comment by ckarande** [comment]
Yes, you are right.

**Comment by binarymist** [comment]
"multiple branches (one per OWASP Top 10 vulnerability) with fixes incorporated in it" is a lot of branches to maintain on every change to master @ckarande

**Comment by ckarande** [comment]
Good point about maintainability @binarymist . Yes, i see it can be a lot of overhead to apply changes to multiple branches. I like the idea of having one defective and another single secure app branch. Having functional tests could help ensuring the ongoing updates to both branches.
